addon/components/model-form/input.hbs

```diff
@@ -0,0 +1,49 @@
+<div class="{{if @horizontal "uk-flex"}}">
+  <div class="uk-text-bold {{if @horizontal "uk-margin-left uk-flex-last"}}">
+    {{component @labelComponent}}
+  </div>
+
+  {{#let (if (eq @type "date") 
+    "model-form/date" 
+    (if (and (eq @type "number") (or @step @min @max))
+      "model-form/step"
+      "validated-input/render"
+    )) as |fieldComponent|
```

Comment:
This is maybe something to handle in js as this looks very confusing in template :sweat_smile: